### DISCUSSION TITLE
Pagination and cache-and-network policy

### DISCUSSION THREAD

**Discussion by Miloshinjo** [urql-graphql/urql#1730] [Q&A] [answered]
How would you handle having a paginated list in the cache/UI with cache and network policy, when user comes back to the list but more items have been added then it was the `limit` or `first`. (Specific: it's cursor pagination, to relay specification).

So if user comes back to a list , but more items have been added, and cache-and-network kicks in, only latest items will be added, and the ones in-between will be eaten up?  Maybe that policy isn't built for that and I should turn to subscriptions in this case? 

Thanks

**Comment by kitten** [comment, marked as answer]
If you can't get up-to-date data via other means then the easiest method would be to continue using `cache-and-network` but to ensure that you start over. As long as your UI takes care of the individual pages (See: https://github.com/FormidableLabs/urql/tree/main/examples/with-pagination) when the page is mounted you'd start at page one again which would ensure that your data is refetched.

The more cache updates you then have to handle changes, the lower the chance is of the user encountering skipped items.

However, overall lists are tricky. Another alternative is to use cursor pagination, like Relay's, with backwards pagination and cursors that include timestamps, but that's obviously much trickier from an API implementation standpoint

**Reply by Miloshinjo** [reply]
Thanks for answering Phil, you are a superhero :)

**Comment by Miloshinjo** [comment]
@kitten Another question. Is there a way to keep subscription connection alive when the component that is using `useSubscription` is unmounted?

**Reply by kitten** [reply]
I mean, you can hoist it higher up, but I'm sure you considered that. The safest bet is to use the `Client` and to use `client.executeSubscription` directly. I obviously also have to warn you that if you're aiming to do this to avoid missing messages then the sad truth is that depending on the network connection you are always subject to missing messages without queuing 😅

**Reply by Miloshinjo** [reply]
Ugh, yea, that can be an issue :) Thanks for answering Phil:)
